### ISSUE TITLE
Work with CodeAnalysis (FxCop, ..)

### ISSUE THREAD

**Issue by Joelius300** [Joelius300/ChartJSBlazor#64]
I think, in general, it would be a great idea to add [Microsoft.CodeAnalysis.Analyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers) and [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers) to get some support for best practices etc.  

However, currently there are some horrible inconsistencies, bad practices and other (style-)issues which won't be fixed until we rework those charts. This means that these analyzers are going to complain all the time and the project will be flooded with (even more) warnings.  

Do we want that?  
In other words, when do we want to add those analyzers?  

Discussions are welcome below.  

CC @SeppPenner

**Comment by SeppPenner** [comment]
I guess we should add this after most of the rework is done.
